File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/clinical/core/domain/HealthInsuranceInfo.java
package io.zenwave360.example.clinicaltool.modules.clinical.core.domain;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import java.io.Serializable;
import java.math.*;
import java.time.*;
import java.util.*;

/**
 *
 */
@lombok.Getter
@lombok.Setter
@Embeddable
public class HealthInsuranceInfo implements Serializable {

    @java.io.Serial
    private static final long serialVersionUID = 1L;

    @NotNull
    @Size(max = 100)
    @Column(name = "insurance_company_id", nullable = false, length = 100)
    private String insuranceCompanyId;

    @NotNull
    @Size(max = 20)
    @Column(name = "insurance_card_number", nullable = false, length = 20)
    private String insuranceCardNumber;
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/clinical/core/domain/HealthInsuranceInfo.kt
package io.zenwave360.example.clinicaltool.modules.clinical.core.domain

import jakarta.persistence.*
import jakarta.validation.constraints.*
import java.io.Serializable
import java.math.*
import java.time.*
import java.util.*

/**  */
@Embeddable
data class HealthInsuranceInfo(
    @NotNull
    @Size(max = 100)
    @Column(name = "insurance_company_id", nullable = false, length = 100)
    var insuranceCompanyId: String? = null,
    @NotNull
    @Size(max = 20)
    @Column(name = "insurance_card_number", nullable = false, length = 20)
    var insuranceCardNumber: String? = null,
) : Serializable {

    companion object {
        private const val serialVersionUID = 1L
    }

    override fun toString(): String {
        return this::class.java.name + "@" + Integer.toHexString(hashCode())
    }
}
